gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/integration/use_case/IngestIntegrationApisUseCase.java

```diff
@@ -90,50 +98,104 @@ public Completable execute(Input input) {
                         var federatedApi = ApiModelFactory.fromIntegration(api, integration);
 
                         if (apiCrudService.existsById(federatedApi.getId())) {
-                            log.debug("API already ingested [id={}] [name={}]", api.id(), api.name());
-                            return;
-                        }
-
-                        createApi(federatedApi, primaryOwner, auditInfo);
-
-                        if (api.plans() != null) {
-                            createPlans(api.plans(), federatedApi, auditInfo);
-                        }
-
-                        if (api.pages() != null) {
-                            createDocumentation(api, federatedApi.getId(), auditInfo);
+                            updateApi(federatedApi, api, auditInfo, primaryOwner);
```

Comment:
Instead of using `existsById` what about fetching the existing API and passing it to the update method. You could use it directly to build the page name and also we could only update the attributes that are coming from the provider (name, description, version) without overriding the other. This would keep the user changes on attribute not managed by Federation agent 🤔

Reply:
```suggestion
                            updateApi(       currentApi
                .toBuilder()
                .name(federatedApi.getName())
                .description(federatedApi.getDescription())
                .version(federatedApi.getVersion()).build(), api, auditInfo, primaryOwner);
```

Reply:
done